pkg/controller/operators/catalog/operator.go

```diff
@@ -606,6 +663,44 @@ func (o *Operator) nothingToUpdate(logger *logrus.Entry, sub *v1alpha1.Subscript
 	return false
 }
 
+func (o *Operator) ensureSubscriptionInstallPlanState(logger *logrus.Entry, sub *v1alpha1.Subscription) (*v1alpha1.Subscription, error) {
+	if sub.Status.Install != nil {
+		return sub, nil
+	}
+
+	logger.Debug("checking for existing installplan")
+
+	// check if there's an installplan that created this subscription (only if it doesn't have a reference yet)
+	// this indicates it was newly resolved by another operator, and we should reference that installplan in the status
+	ips, err := o.client.OperatorsV1alpha1().InstallPlans(sub.GetNamespace()).List(metav1.ListOptions{})
```

Comment:
nit: no lister?

Reply:
there wasn't an installplan lister to use, I forgot to go back and add one. I'll update.